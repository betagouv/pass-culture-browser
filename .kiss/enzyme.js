import React from 'react'
import { shallow } from 'enzyme'

import MyComponent from '../MyComponent'

describe('src | components | MyComponent', () => {
  describe('snapshot', () => {
    it('should match snapshot', () => {
      const props = { errors: {} }
      const wrapper = shallow(<MyComponent {...props} />)
      expect(wrapper).toBeDefined()
      expect(wrapper).toMatchSnapshot()
    })
  })
})
