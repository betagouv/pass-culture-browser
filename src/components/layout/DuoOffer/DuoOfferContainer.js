import { connect } from 'react-redux'

import DuoOffer from './DuoOffer'
import selectIsFeatureDisabled from '../../router/selectors/selectIsFeatureDisabled'
import { selectIsEnoughStockForOfferDuo } from '../../../selectors/data/stocksSelector'
import selectOfferById from '../../../selectors/selectOfferById'

export const mapStateToProps = (state, ownProps) => {
  const { offerId } = ownProps
  const isFeatureDisabled = selectIsFeatureDisabled(state, 'DUO_OFFER')
  const isEnoughStockForDuo = selectIsEnoughStockForOfferDuo(state, offerId)
  const offer = selectOfferById(state, offerId)
  const { isDuo } = offer
  const isDuoOffer = !isFeatureDisabled && isEnoughStockForDuo && isDuo

  return {
    isDuoOffer,
  }
}

export default connect(mapStateToProps)(DuoOffer)
