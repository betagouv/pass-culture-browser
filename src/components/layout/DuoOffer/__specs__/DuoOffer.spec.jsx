import React from 'react'
import { shallow } from 'enzyme'

import DuoOffer from '../DuoOffer'

describe('src | components | layout | DuoOffer | DuoOffer', () => {
  it('when this offer is a not an "offre duo" should render nothing', () => {
    // given
    const props = {
      isDuoOffer: true,
      label: 'Vous pouvez réserver deux places.',
      offerId: 'AFZR',
    }

    // when
    const wrapper = shallow(<DuoOffer {...props} />)

    // then
    expect(wrapper).toMatchSnapshot()
  })

  it('when this offer is an "offre duo" should render picto and label', () => {
    // given
    const props = {
      isDuoOffer: false,
      label: 'Vous pouvez réserver deux places.',
      offerId: 'ABCD',
    }

    // when
    const wrapper = shallow(<DuoOffer {...props} />)

    // then
    expect(wrapper).toMatchSnapshot()
  })
})
