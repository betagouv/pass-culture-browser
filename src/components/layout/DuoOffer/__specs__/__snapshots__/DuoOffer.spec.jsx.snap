// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`src | components | layout | DuoOffer | DuoOffer when this offer is a not an "offre duo" should render nothing 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DuoOffer
    isDuoOffer={true}
    label="Vous pouvez réserver deux places."
    offerId="AFZR"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <span
          className="duo-logo"
        >
          duo
        </span>,
        <span
          className="duo-label"
        >
          Vous pouvez réserver deux places.
        </span>,
      ],
      "className": "duo",
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "duo",
          "className": "duo-logo",
        },
        "ref": null,
        "rendered": "duo",
        "type": "span",
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": "Vous pouvez réserver deux places.",
          "className": "duo-label",
        },
        "ref": null,
        "rendered": "Vous pouvez réserver deux places.",
        "type": "span",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <span
            className="duo-logo"
          >
            duo
          </span>,
          <span
            className="duo-label"
          >
            Vous pouvez réserver deux places.
          </span>,
        ],
        "className": "duo",
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "duo",
            "className": "duo-logo",
          },
          "ref": null,
          "rendered": "duo",
          "type": "span",
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": "Vous pouvez réserver deux places.",
            "className": "duo-label",
          },
          "ref": null,
          "rendered": "Vous pouvez réserver deux places.",
          "type": "span",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;

exports[`src | components | layout | DuoOffer | DuoOffer when this offer is an "offre duo" should render picto and label 1`] = `
ShallowWrapper {
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <DuoOffer
    isDuoOffer={false}
    label="Vous pouvez réserver deux places."
    offerId="ABCD"
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "checkPropTypes": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateError": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): false,
  Symbol(enzyme.__nodes__): Array [
    false,
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
        "legacyContextMode": "parent",
        "lifecycles": Object {
          "componentDidUpdate": Object {
            "onSetState": true,
          },
          "getChildContext": Object {
            "calledByRenderer": false,
          },
          "getDerivedStateFromError": true,
          "getDerivedStateFromProps": Object {
            "hasShouldComponentUpdateBug": false,
          },
          "getSnapshotBeforeUpdate": true,
          "setState": Object {
            "skipsComponentDidUpdateOnNullish": true,
          },
        },
      },
    },
    Symbol(enzyme.__providerValues__): undefined,
  },
  Symbol(enzyme.__providerValues__): Map {},
}
`;
